clients/src/main/java/org/apache/kafka/clients/producer/internals/BufferPool.java

```diff
@@ -50,7 +50,7 @@
     private final ReentrantLock lock;
     private final Deque<ByteBuffer> free;
     private final Deque<Condition> waiters;
-    /** This memory is accounted for separately from the poolable buffers in free. */
+    /** Poolable + non-poolable memory. */
     private long availableMemory;
```

Comment:
If we change the definition of the available memory here, we may have to change a few other places as well. e.g. `deallocate()`, `unallocatedMemory()`, etc.